### DISCUSSION TITLE
iperf_run_tcp_client(361): Socket unable to connect: errno 113

### DISCUSSION THREAD

**Discussion by arvind55555** [esnet/iperf#1601] [Q&A] [answered]
hi everyone,
im currently trying to run iperf example using esp32 pico V4. i was getting lot of issues but finally i made my UDP work. 
![UDP WORKING SS](https://github.com/esnet/iperf/assets/139859655/8233dfd6-74bb-46bb-bc30-5cfd3e8f2872)

initially UDP throughput didn't work for me in STA mode, but now (idk how) UDP is working in STA mode.
![STA CONNECTION](https://github.com/esnet/iperf/assets/139859655/22b98ee1-f751-42ae-bfd3-9face3dad9df)

the only issue that i have been facing for a long time now is that i am unable to make the TCP throughput work. I tried every solution that i could get my hands on, but none of them works. i know that it should be something very very simple, that i missing out.
![TCP NOT WORKING](https://github.com/esnet/iperf/assets/139859655/020d7c1b-5ca0-4324-afa0-3de5c9cf6124)

please do help me out!!!

**Comment by bmah888** [comment]
This doesn't look like iperf3. I would have guessed this was iperf2 (a completely separate project with a different code repository and issue tracker), but I'm unable to match the messages in your output above with strings in the iperf2 source code. What does the output of `iperf --version` give you?

**Reply by arvind55555** [reply]
hi thanks for responding,
![iperf --version](https://github.com/esnet/iperf/assets/139859655/068aea70-7ef9-4f75-a8fa-3f01a6603220)

this is what i get when i type iperf --version.

i'm actually running this example on **ESP-IDF 5.1 CMD**

**Reply by davidBar-On** [reply]
As this is an iperf3 discussion forum, it is probably not the right place for the question.  Searching the internet for "ESP-IDF iperf errno 113" there are several results.  Suggest to look at these answers and use one of their forums for the question.

**Reply by bmah888** [reply, marked as answer]
^^^ This. Or, if there's some kind of official tech support mechanism for ESP-IDF, try that. This really appears to be a customized variant of a different form of iperf, embedded in some other product. It's not real likely that anyone working on iperf3 would be able to help you. Good luck!

**Reply by arvind55555** [reply]
Thanks for y'all help
